### PR TITLE
WIP: Add CRRL/CRAM

### PR DESCRIPTION
**Pull request by arichardson**
CCheckPerm/CCheckType have been removed and those opcodes are used for
CRRL/CRAM. This is mostly a copy of the MIPS implementation but I can't
get the sail typechecker to accept my changes.
Pushing this as a PR now so I don't forget

### CODE REVIEW

File: src/cheri_insts.sail [hunk 1]
```diff
@@ -1155,8 +1132,8 @@ mapping clause assembly = CClearTag(cd, cs)  <-> "CClearTag"  ^ spc() ^ reg_name
 mapping clause assembly = CJALR(0b00000, cb) <-> "CJR"        ^ spc() ^ reg_name(cb)
 mapping clause assembly = CJALR(cd, cb)      <-> "CJALR"      ^ spc() ^ reg_name(cd) ^ sep() ^ reg_name(cb)
 
-mapping clause assembly = CCheckPerm(cs, rt) <-> "CCheckPerm" ^ spc() ^ reg_name(cs) ^ sep() ^ reg_name(rt)
-mapping clause assembly = CCheckType(cs, cb) <-> "CCheckType" ^ spc() ^ reg_name(cs) ^ sep() ^ reg_name(cb)
+mapping clause assembly = CRLL(rd, rs) <-> "CRRL" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs)
```
Comment (bauereiss) [inline src/cheri_insts.sail]: There's a typo here:
```suggestion
mapping clause assembly = CRRL(rd, rs) <-> "CRRL" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs)
```

**Comment by rmn30** [comment]
I like the addition of ie_exponent_take_bits but the name is a bit wordy. Could also use it in a few other places to get rid of magic 3s.

**Comment by arichardson** [comment]
> I like the addition of ie_exponent_take_bits but the name is a bit wordy. Could also use it in a few other places to get rid of magic 3s.

Yeah I couldn't think of a better name
